packages/backend/src/matter/devices/media-player-device.ts

```diff
@@ -1,17 +1,51 @@
-import { MatterDevice } from "../matter-device.js";
-import { OnOffPlugInUnitDevice } from "@matter/main/devices";
-import { OnOffServer } from "../behaviors/on-off-server.js";
+import {
+  HomeAssistantEntityState,
+  SpeakerDeviceAttributes,
+} from "@home-assistant-matter-hub/common";
+import { SpeakerDevice } from "@matter/main/devices";
 import { BasicInformationServer } from "../behaviors/basic-information-server.js";
 import { IdentifyServer } from "../behaviors/identify-server.js";
 import { HomeAssistantBehavior } from "../custom-behaviors/home-assistant-behavior.js";
+import { OnOffServer } from "../behaviors/on-off-server.js";
+import { MatterDevice } from "../matter-device.js";
+import {
+  LevelControlConfig,
+  LevelControlServer,
+} from "../behaviors/level-control-server.js";
+import { SwitchDevice } from "./switch-device.js";
+
+const volumeLevelConfig: LevelControlConfig = {
+  getValue: (state: HomeAssistantEntityState<SpeakerDeviceAttributes>) => {
+    if (state.attributes.volume_level != null) {
+      return state.attributes.volume_level * 254;
+    }
+    return 0;
+  },
+  moveToLevel: {
+    action: "media_player.volume_set",
+    data: (value) => {
+      return { volume_level: value / 100 };
+    },
+  },
+};
 
-const MediaPlayerEndpointType = OnOffPlugInUnitDevice.with(
+const MediaPlayerEndpointType = SpeakerDevice.with(
   BasicInformationServer,
   IdentifyServer,
   HomeAssistantBehavior,
   OnOffServer,
+  LevelControlServer,
 );
 
 export function MediaPlayerDevice(homeAssistant: HomeAssistantBehavior.State) {
-  return new MatterDevice(MediaPlayerEndpointType, homeAssistant);
+  switch (homeAssistant.entity.attributes.device_class) {
```

Comment:
The `switch` is arguable here.. but I figured in the future we could create different matter devices/behaviours based on `device_class`.

Including `tv` is even more arguable.. I tried to play with `CastingVideoPlayer` and even though it shows as a TV in Google Home it doesn't allow volume control. 

What do you think?

Reply:
I‘m fine with it, but I am not a huge fan of switch case. Have a look at binary_sensor to see how I did it.

Reply:
Anyway you didn’t add the device class to the speaker attributes. And maybe you could introduce an enum with all values.

Reply:
How would you feel about something like this? (this is just and draft but you'll get the gist)

```
const baseBehaviors = [
  IdentifyServer,
  BasicInformationServer,
  HomeAssistantBehavior,
  OnOffServer,
];

const deviceMapping: Record<MediaPlayerDeviceClass, MutableEndpoint> = {
  [MediaPlayerDeviceClass.Tv]: CastingVideoPlayerDevice,
  [MediaPlayerDeviceClass.Speaker]: SpeakerDevice,
  [MediaPlayerDeviceClass.Receiver]: CastingVideoPlayerDevice,
};
// Mapping overridable by config

export function MediaPlayerDevice(homeAssistant: HomeAssistantBehavior.State) {
  const deviceClass: MediaPlayerDeviceClass =
    homeAssistant.entity.attributes.device_class;

  const matterDevice: MutableEndpoint = deviceMapping[deviceClass];
  const matterBehaviors = [...baseBehaviors];
  const config = {};

  const supportedFeatures =
    homeAssistant.entity.attributes.supported_features || 0;

  // Constants for volume control features
  const SUPPORT_VOLUME_SET = 4;
  const SUPPORT_VOLUME_STEP = 1024;
  const hasVolumeControl =
    supportedFeatures & SUPPORT_VOLUME_SET ||
    supportedFeatures & SUPPORT_VOLUME_STEP;

  if (hasVolumeControl) {
    matterBehaviors.push(LevelControlServer);
   config.levelControl = { config: volumeLevelConfig };
  }
  // add other behaviors for playback control, streaming, etc.

  const type = matterDevice.with(...matterBehaviors);
  return new MatterDevice(type, homeAssistant, config);
}
```

In my opinion, it's a bit more flexible and forward-looking, which should be helpful if and when such controls are eventually supported.

Reply:
tbh I’d like to keep the config simple. Therefore I’d prefer compatibility mode over detailed configs. But I get your point. And I will somehow need a more detailed way to feature-flag things.